File: jscomp/test/polymorphism_test.js
'use strict';


function map(f, param) do
  if (param) do
    var r = f(param[0]);
    return --[ :: ]--[
            r,
            map(f, param[1])
          ];
  end else do
    return --[ [] ]--0;
  end
end

exports.map = map;
--[ No side effect ]--

File: jscomp/test/polymorphism_test.lua
__console = {log = print};


function map(f, param) do
  if (param) then do
    r = f(param[1]);
    return --[[ :: ]]{
            r,
            map(f, param[2])
          };
  end else do
    return --[[ [] ]]0;
  end end 
end end

exports = {};
exports.map = map;
return exports;
--[[ No side effect ]]
